Properly handle exceptions when re-parsing with dummy sources

nforro opened the pull request:
Fixes #172.

TomasTomecek reviewed (commented):
would it make sense to include the excerpt of the grub2 spec file in tests to account for this use case?

nforro commented:
> would it make sense to include the excerpt of the grub2 spec file in tests to account for this use case?

I can make a test case for this.

mfocko commented on specfile/spec_parser.py:
```diff
@@ -250,19 +252,20 @@ def collect_included_sources(content):
-                        # non empty sources (if any), remove them
-                        sources = {s for s, _, _ in spec.sources} - non_empty_sources
+                    if not dummy_sources:
+                        raise
```
That made me a think for a moment :D

mfocko reviewed (commented):
+1 after the tests

nforro commented:
I've added a test case and I've also added support for `%{load:...}`, which is an equivalent of `%include ...`, but only for macro files (where macros are defined directly without `%global` or `%define`).

TomasTomecek reviewed (approved):
👏
